foundation-models/openai/src/main/java/com/sap/ai/sdk/foundationmodels/openai/OpenAiChatCompletionRequest.java

```diff
@@ -282,6 +282,19 @@ public OpenAiChatCompletionRequest withToolChoice(@Nonnull final OpenAiToolChoic
     return this.withToolChoice(choice.toolChoice);
   }
 
+  /**
+   * Sets the tools to be used in the request with convenience class {@code OpenAiTool}.
+   *
+   * @param tools the list of tools to be used
+   * @return a new OpenAiChatCompletionRequest instance with the specified tools
+   * @throws IllegalArgumentException if the tool type is not supported
+   * @since 1.7.0
+   */
+  @Nonnull
+  public OpenAiChatCompletionRequest withOpenAiTools(@Nonnull final List<OpenAiTool<?>> tools) {
```

Comment:
(Major)

There currently is `withTools( List<ChatCompletionTool> )` for low-level/generated API.

You want to add `withOpenAiTools( List<OpenAiTool<?>> )` for high-level/convenience API.

The problem is they don't have common prefix.  Users will not know what method to use for tools. This feels inconsistent to them.

Reply:
I changed it

```diff
- withTools( List<ChatCompletionTool> )
+ withToolsExecutable( List<OpenAitool<?>> )
```